### ISSUE TITLE
CI: Downstream-integration tests are failing

### ISSUE THREAD

**Issue by csadorf** [aiidalab/aiidalab-qe#205]
The [_scheduled_ downstream-integration workflow](https://github.com/aiidalab/aiidalab-qe/actions/workflows/di.yml) is currently failing. 

@unkcpz I would recommend to fix it and disable the workflow until then.

**Comment by csadorf** [comment]
Yes, we probably need to revise the action. Without any additional test scripts, it primarily acts as a smoke test, where it tries to open the app in the browser and checks that it loaded without any issues. This will catch 80% of critical failure cases related to dependency incompatibilities and so on.

**Comment by unkcpz** [comment]
Mmore subtle issue about downstream CI test. The qe-app can't be loaded by using the `[aiidalab-test-app-action](https://github.com/aiidalab/aiidalab-test-app-action)`.

**Comment by csadorf** [comment]
The goal is to be able to run `pytest` inside the repository root, and it will automatically run all tests.

Proposed approach for implementing tests:
1. Copy the approach for setting up a test environment from the [aiidalab-docker-stack test config](https://github.com/aiidalab/aiidalab-docker-stack/blob/main/tests/conftest.py) with [pytest-docker](https://pypi.org/project/pytest-docker/) (test against `docker.io/aiidalab/full-stack:edge`).
2. Install the tested app by mounting the repository (/the GitHub workspace directory) and adding it to the AIIDALAB_DEFAULT_APPS environment variable (see example below).
3. Implement basic smoke tests (is app directory present, is app reported installed by aiidalab etc.)
4. Implement interface tests with selenium, you can get the url from the [`notebook_service` fixture](https://github.com/aiidalab/aiidalab-docker-stack/blob/7f89fde47696ce2d80e1c959da5a1181fe2b9a27/tests/conftest.py#L30).

```yml
---
# tests/docker-compose.yml
version: '3.4'
services:
    aiidalab:
        image: aiidalab/full-stack:edge
        ports:
            - 8888
        volumes:
            - ${PWD}:/home/jovyan/workspace
        env:
            AIIDALAB_DEFAULT_APPS: quantum-espresso@file:///home/jovyan/workspace
```

**Comment by unkcpz** [comment]
This will be fixed by #271 where I implement the basic test along the way suggested.
